test/icinga-legacytimeperiod.cpp

```diff
@@ -211,6 +211,36 @@ BOOST_AUTO_TEST_CASE(simple)
 
 	BOOST_CHECK_EQUAL(begin, expectedBegin);
 	BOOST_CHECK_EQUAL(end, expectedEnd);
+
+	//-----------------------------------------------------
+	tm_beg = Utility::LocalTime(1706518800); // 2024-01-29 10:00:00
+	tm_end = Utility::LocalTime(1706520600); // 2024-01-29 10:30:00
+
+	tm reference = Utility::LocalTime(1706519400); // 2024-01-29 10:10:00
+	reference.tm_sec = 0;
+	reference.tm_isdst = -1;
+
+	// The reference time is only 10 minutes ahead of the start date, which should be covered by this range.
+	BOOST_CHECK_EQUAL(true, LegacyTimePeriod::IsInTimeRange(&tm_beg, &tm_end, 1, &reference));
+
+	reference = Utility::LocalTime(1706520600); // 2024-01-29 10:30:00
+
+	// The reference time is exactly the same as the specified end time, so this should definitely not be in the range.
+	BOOST_CHECK_EQUAL(false, LegacyTimePeriod::IsInTimeRange(&tm_beg, &tm_end, 1, &reference));
```

Comment:
That's nice, but I'd prefer four tests. Two edge cases at each end of the range.

* 2024-01-29 09:59:59 false
* 2024-01-29 10:00:00 true
* 2024-01-29 10:29:59 true
* 2024-01-29 10:30:00 false

While on it, I'd also prefer you to make a separate BOOST_AUTO_TEST_CASE for these four and to push just it. Once GHA is done and clearly fails, please push the actual change.

Reply:
> Once GHA is done and clearly fails, please push the actual change.

I'm not going to do this now! If you have doubts about the fix, you can also build Icinga 2 locally and run the tests there. I don't to waste my time with such nonsense!